### ISSUE TITLE
Staking Adding `destination` field to `Rewarded` event to allow easier reward tracking

### ISSUE THREAD

**Issue by rossbulat** [paritytech/substrate#14792]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Request

Teams have began exploring a better mechanism of estimating staking rewards, which inevitably involves indexing chain state and collecting a historical amount of data. 

It is becoming apparent that this is expensive and cumbersome to do with the current block data being emitted. Notably, it is very non-trivial to track to what address(es) staking rewards are actually sent to: the final destination, rather than the stash account subject to the reward.

To further complicate this tracking, the [upcoming `Split`](https://github.com/paritytech/substrate/pull/14451) payout destination will allow rewards to go to potentially multiple addresses, one part to compound bond, the other part as free balance.

One potential way we can streamline these issues is to have a `destination` field in the `Rewarded` event, emitted when payouts are made to stakers (validators & nominators). Another suggestion has been just to include the final payout destination address, but again, the `Split` variant complicates this as there will most likely 2 distinct addresses in most cases. Including the `RewardDestination`(or upcoming `PayoutDestination` in the event would give clients tracking the event enough information to successfully track exactly where staking rewards are going to, in real-time, as the event is emitted.

Beyond an additional space requirement for the additional field, I can not foresee any unwanted impacts by adding this field. It would aid in tracking reward data, and ultimately lead to better means of estimating APY (which we very much need in place of the rough estimations we are using).

### Are you willing to help with this request?

Yes!

**Comment by Ank4n** [comment]
To add to this, there are up to 512 nominators paid in a single call instruction. The rewarded account can be inferred today by reading the `Payee` storage item but it needs to be read via RPC call for each of these (~512) rewarded events 

I support the idea of explicit addition of which account/s has been `Rewarded` to the event. 

Should mention though that `payout_stakers` is already a heavy operation, and adding a new field to this event would make it (additional cost of a new field * 512) more expensive. Once we have paged rewards, hopefully we could reduce the number of payouts in a single call.

**Comment by Ank4n** [comment]
To add to this, the target account can be inferred today by reading the `Payee` storage item for each `Rewarded` event (which is a lot of RPC calls).

I am in support of adding explicitly which account has been rewarded, seems like a important information missing from the event. Though I should mention that `payout_stakers` is already a heavy operation and this would make it more expensive by (cost of adding a field * 512). Hopefully, once we have paged rewards, we could reduce the number of payouts in a single call.

**Comment by rossbulat** [comment]
Indeed, I feel a lot is hanging on the paged rewards PR, for this and for the split payout destination. I have no strong option about whether to hold off until paged payouts are merged or not, but by intuition is that as long as `payout_stakers` still leaves a modest amount of space for other block data, we should be good.

**Comment by joepetrowski** [comment]
> which is a lot of RPC calls

We already have tools like Sidecar that aggregate these calls so that users (front ends, custodians, etc.) can make a single query and get all this info. Like you said it's already a heavy on-chain operation, and referencing only the stash makes sense (that is what generated the rewards). Other information can easily be looked up with the stash account as key.
